clients/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceAfterImageConsumerImpl.java

```diff
@@ -38,11 +47,21 @@
 
   private boolean isReadFromChangeCaptureTopic;
 
+  private final CompressorFactory compressorFactory = new CompressorFactory();
+  private VeniceCompressor currentCompressor = NO_OP_COMPRESSOR;
+
   public VeniceAfterImageConsumerImpl(
       ChangelogClientConfig changelogClientConfig,
       Consumer<KafkaKey, KafkaMessageEnvelope> kafkaConsumer) {
     super(changelogClientConfig, kafkaConsumer);
     this.currentTopic = Version.composeKafkaTopic(storeName, storeCurrentVersion);
+    Class<V> valueClass = changelogClientConfig.getInnerClientConfig().getSpecificValueClass();
```

Comment:
So far inside `VeniceChangelogConsumerClientFactory` and `TestActiveActiveIngestion`, we have Raw use of parameterized class 'ChangelogClientConfig', that means here the `valueClass` is always null. So do we plan to pass real valueClass inside Factory?

Reply:
That is correct.  I was planning on making it available for folks who would prefer to work with specific record to improve performance.  No one has asked for this yet though.

Reply:
Hrm.  Though I see an issue here with the way the clientfactory is set up.  The interface just takes store name today.  We should add one which also takes a value class, otherwise this interface won't work very well.

Reply:
SGTM! Do you plan to do that in this change?

Reply:
No, let me do it in another, this PR is quite large at this point.